api/users.js

```diff
@@ -0,0 +1,277 @@
+const usersRouter = require("express").Router();
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
+const {
+  createUser,
+  updateUser,
+  getUserByUserId,
+  getUserByUsername,
+  getAllUsers,
+  getAllMerchandise,
+  getUserPreferencesByUserId,
+  createUserPreferences,
+} = require("../db");
+const { requireUser, requireActiveUser } = require("./utils");
+// const { useLayoutEffect } = require('react');
+
+usersRouter.use((req, res, next) => {
+  console.log("A request is being made to /users");
+  next();
+});
+
+usersRouter.get("/", async (req, res) => {
+  const users = await getAllUsers();
+  res.send({
+    users,
+  });
+});
+
+usersRouter.get("/me", requireUser, async (req, res, next) => {
+  const user = req.user;
+  console.log("User Object: ", user);
+
+  if (user && user.user_id) {
+    try {
+      res.send({
+        message: "Here is the fetched user!",
+        user: await getUserByUserId(user.user_id),
+        status: true,
+      });
+    } catch ({ name, message }) {
+      next({ name, message });
+    }
+  }
+});
+
+usersRouter.post("/register", async (req, res, next) => {
+  const {
+    username,
+    password,
+    firstname,
+    lastname,
+    street,
+    city,
+    state,
+    zip,
+    save_pmt,
+    shipping,
```

Comment:
Where is all of this information going to go? None of the db tables are set up to use this info.